### PR TITLE
angular 9 upgrade

### LINKED ISSUE #217: Angular 9 upgrade
upgrade styleguide component library to Angular 9

### PR DESCRIPTION
**Pull request by sandikbarr**
closes #217

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
         "build": {
           "builder": "@angular-devkit/build-angular:browser",
           "options": {
+            "aot": true,
```
Comment (sandikbarr) [inline angular.json:16]: CLI apps compile in AOT mode by default

File: bin/buildwatch.js [hunk 2]
```diff
@@ -2,83 +2,73 @@ const sane = require('sane');
...
+    if (containerId) {
       console.log(`Found running ${CONTAINER_NAME} container: ${containerId}`);
-      exec(`docker cp dist/ngx-prx-styleguide/. ${containerId}:/app/node_modules/ngx-prx-styleguide`)
-        .then(({stdout,stderr}) => {
-          if(stderr) {
-            console.log(stderr);
-          }
-          console.log('copied successfully');
-          console.log(stdout);
-        });
```
Comment (sandikbarr) [inline bin/buildwatch.js:60]: Sorry about all the formatting changes and not making a separate commit. Change is here: It didn't work to just copy it over anymore, bunch of errors. So now it uses `npm pack` to tar up the build in the dist directory then it does a copy and exec's an install on the docker container.

File: projects/ngx-prx-styleguide/src/lib/fancy-form/padzero.pipe.stories.ts [hunk 3]
```diff
@@ -50,7 +48,7 @@ __Name__ \`padzero\`
 
 ## Usage
 \`\`\`html
-<span>{{ intNumber | capitalize:intPadLength }}</span>
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/lib/fancy-form/padzero.pipe.stories.ts:53]: typo

File: projects/ngx-prx-styleguide/src/lib/hero/hero.component.spec.ts [hunk 4]
```diff
@@ -48,8 +48,8 @@ describe('HeroComponent', () => {
   });
 
   const triggerScroll = () => {
-    let e = document.createEvent('UIEvents');
-    e.initUIEvent('scroll', true, true, window, 1);
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/lib/hero/hero.component.spec.ts:52]: I think we can actually deprecate the ol' hero, but I did not.

File: projects/ngx-prx-styleguide/src/lib/status-bar/status-bar-link/status-bar-link.component.ts [hunk 5]
```diff
@@ -13,9 +13,9 @@ export class StatusBarLinkComponent implements AfterViewInit {
 
   @HostBinding('class.align-art--right') alignArtRight = false;
 
-  @ViewChild('icon', { static: false }) icon: ElementRef;
-  @ViewChild('image', { static: false }) image: ElementRef;
-  @ViewChild('text', { static: false }) text: ElementRef;
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/lib/status-bar/status-bar-link/status-bar-link.component.ts:18]: automated migration, now false by default

File: projects/ngx-prx-styleguide/src/lib/tooltip/tooltip-content.component.ts [hunk 6]
```diff
@@ -0,0 +1,194 @@
+import { Component, Input, AfterViewInit, ElementRef, ChangeDetectorRef } from '@angular/core';
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/lib/tooltip/tooltip-content.component.ts:1]: This came from the tooltip library, which didn't have any tests, and I'm not writing any. The only place this is used is on the adfree quick tag in Publish.

File: projects/ngx-prx-styleguide/src/lib/tooltip/tooltip.directive.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+  @Input()
+  tooltipAnimation = true;
+
+  @Input()
+  tooltipPlacement: 'top' | 'bottom' | 'left' | 'right' = 'bottom';
+
+  @HostListener('focusin')
+  @HostListener('mouseenter')
+  show(): void {
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/lib/tooltip/tooltip.directive.ts:35]: uses standard HostListener stuff with ComponentFactoryResolver to create a component that actually shows the content string

File: projects/ngx-prx-styleguide/src/lib/upload/upload.component.spec.ts [hunk 8]
```diff
@@ -6,36 +6,30 @@ import { HalService } from '../hal/hal.service';
 import { MockHalService } from '../hal/mock/mock-hal.service';
 import { By } from '@angular/platform-browser';
 
-@Pipe({name: 'capitalize'})
-class DurationStubPipe implements PipeTransform {
+@Pipe({ name: 'capitalize' })
+class CapitilizeStubPipe implements PipeTransform {
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/lib/upload/upload.component.spec.ts:10]: just a misnamed pipe and formatting in this file ;)

File: projects/ngx-prx-styleguide/src/public_api.ts [hunk 9]
```diff
@@ -1,29 +1,47 @@
 export { ChartsModule } from './lib/charts/charts.module';
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/public_api.ts:1]: a change in the public_api: It now needs to include all components that are exported by the ng modules. I initially tried to use barrel files and reduce the amount of import statements, but that didn't work for the library to be consumed by applications. It's never a fun time to guess that it's barrel files that are causing you errors and headaches, so I also went around a couple places and totally got rid of em.

File: projects/ngx-prx-styleguide/src/public_api.ts [hunk 10]
```diff
@@ -37,39 +55,81 @@ export { HalHttpError } from './lib/hal/remote/halremote';
...
 export { ToastrService } from './lib/toastr/toastr.service';
 
-
-export { UploadModule } from './lib/upload/upload.module';
 export { PlayerService } from './lib/audio';
-export { MimeTypeService, UploadService, Upload } from './lib/upload/service';
-export { AudioFileModel, AudioVersionModel, AudioVersionTemplateModel, AudioFileTemplateModel } from './lib/upload/model';
-export { UploadableModel, HasUpload, createGetUploads, createSetUploads } from './lib/upload/model/upload';
-export { DurationPipe, FileSelectDirective, FileSizePipe } from './lib/upload/file';
+export * from './lib/upload/file/duration.pipe';
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/public_api.ts:108]: hate to say it after how hard it was for Peter to move this all over here, but this should probably be moved back to publish if some kind of resuable uploader that doesn't use BaseModel can't be extracted out of it

File: projects/ngx-prx-styleguide/tsconfig.lib.prod.json [hunk 11]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/tsconfig.lib.prod.json:1]: angular 9 libraries should not use Ivy https://indepth.dev/the-angular-ivy-guide-for-library-authors/

File: projects/ngx-prx-styleguide/src/lib/upload/file/index.ts [hunk 12]
```diff
@@ -1,3 +0,0 @@
-export * from './duration.pipe';
```
Comment (cavis) [inline projects/ngx-prx-styleguide/src/lib/upload/file/index.ts:1]: Are these exports not used anywhere?

Ah, I see you removed a bunch of the barrel files.  So maybe they weren't necessary.

File: projects/ngx-prx-styleguide/src/lib/upload/free/free-reorder.directive.ts [hunk 13]
```diff
@@ -19,21 +18,21 @@ export class FreeReorderDirective extends DragulaDirective implements OnInit, On
...
       moves: (el: Element, source: Element, handle: Element) => {
         return handle.classList.contains('drag-handle');
       }
     });
-    this.dragSub = this.myDragula.dropModel.subscribe(() => {
+    this.dragSub = this.myDragula.dropModel(groupName).subscribe(() => {
       this.prxFreeReorder.reassign();
     });
-    super.ngOnInit();
+    // super.ngOnInit();
```
Comment (cavis) [inline projects/ngx-prx-styleguide/src/lib/upload/free/free-reorder.directive.ts:30]: Hrm, yeah, that probably wasn't doing anything.

File: tsconfig.json [hunk 14]
```diff
@@ -5,7 +5,7 @@
     "outDir": "./dist/out-tsc",
     "sourceMap": true,
     "declaration": false,
-    "module": "es2015",
+    "module": "esnext",
```
Comment (cavis) [inline tsconfig.json:8]: Latest and greatest?

File: projects/ngx-prx-styleguide/src/lib/upload/file/index.ts (hunk 12 above)
Comment (cavis) [reply projects/ngx-prx-styleguide/src/lib/upload/file/index.ts:1]: Ah... and reading the rest of the PR, I see these were causing problems.  +100!

**Review by cavis — APPROVED** [review]
👍 Looks good to me!  Just needs the ol' version bump.

File: projects/ngx-prx-styleguide/src/lib/tooltip/tooltip.directive.ts (hunk 7 above)
Comment (sandikbarr) [reply projects/ngx-prx-styleguide/src/lib/tooltip/tooltip.directive.ts:35]: and not a bad small example for creating ad-hoc components 😸

File: tsconfig.json (hunk 14 above)
Comment (sandikbarr) [reply tsconfig.json:8]: updated with the auto `ng update` migration

**Pull request merged**